thanks-bot/bot.js

```diff
@@ -3,33 +3,67 @@ var logger = require('winston');
 var auth = require('./auth.json');
 
 // Configure logger settings
-logger.remove(logger.transports.Console);
-logger.add(logger.transports.Console, {
-    colorize: true
+logger.remove(logger.transports.Console, {
+  colorize: true
 });
-
 logger.level = 'debug';
 
+// Канал, в котором бот может реагировать на сообщения
+const ALLOWED_CHANNEL_ID = "525249065039036426";
+
 // Initialize Discord Bot
 var bot = new Discord.Client({
-   token: auth.token,
-   autorun: true,
+  token: auth.token,
+  autorun: true
 });
 
-bot.on('ready', function (evt) {
-    logger.info('Connected');
-    logger.info('Logged in as: ');
-    logger.info(bot.username + ' - (' + bot.id + ')');
+bot.on('ready',  function(evt) {
+  logger.info('Connected');
+  logger.info('Logged in as: ');
+  logger.info(bot.username + ' - (' + bot.id + ')');
 });
 
 bot.on('message', function (user, userID, channelID, message, evt) {
-    switch(message) {
-        case 'Ассаляму аляйкум':
-            bot.sendMessage({
-                to: channelID,
-                message: 'Уа аляйкум салям',
-            });
-
-            break;
-    }
+  // в каком канале пишут?
+  if (channelID !== ALLOWED_CHANNEL_ID) {
+    return;
+  }
+
+  var regexp = /\!thanks\b/gmi;
```

Comment:
Move this expression right before `if (!regexp.test(message))`. Additionally, you don't need the `\`
```suggestion
  var regexp = /!thanks\b/gmi;
```